### DISCUSSION TITLE
Time IDs in SDXL training

### DISCUSSION THREAD

**Discussion by christopher5106** [huggingface/diffusers#9459] [General]
I was trying to understand these [time ids](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py#L1093) in SDXL training script. It's an array for each example, that could be [1024, 1034, 0, 10, 1024, 1024]. It's directly fed into [Timesteps forward](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_2d_condition.py#L980). I'm just wondering what's happening here, how each of these integers original_height, original_width, crop_coord_top, crop_coord_left, target_resolution, target_resolution get a kind of exponential embedding... and what is the expected output of this, there is something I'm missing, thanks for your help !

**Comment by asomoza** [comment]
cc: @sayakpaul

**Comment by christopher5106** [comment]
Important to me, because I'm training a classifier on these latents, for classifier guidance, but I can only train it on crops (because i don't have full data). Then I need to be sure that when I apply the classifier to the whole image, time ids will be the same as in the crop if at the same place in the image. 
From reading the code, I can't infer what they do exactly.

**Comment by sayakpaul** [comment]
These are micro-conditions that were introduced in the SDXL paper. You could read more about them in the Section 2.2 of the paper: https://arxiv.org/abs/2307.01952.
